src/ln/channel.rs

```diff
@@ -627,6 +627,27 @@ impl Channel {
 		channel_monitor.set_their_base_keys(&msg.htlc_basepoint, &msg.delayed_payment_basepoint);
 		channel_monitor.set_their_to_self_delay(msg.to_self_delay);
 
+		let their_shutdown_scriptpubkey = if their_local_features.supports_upfront_shutdown_script() {
```

Comment:
Shouldn't we use the shutdown_scriptpubkey if it is Present (ad not-0-length) even if they have't set their local features? We just reject the message if they did set local features (and we did too) and they didnt send a shutdown_scriptpubkey?

Reply:
On first point, spec said "otherwise:

    MAY include ashutdown_scriptpubkey."

But that's a option offered for sending node, not a requirement for the receiving node to enforce upfront_shutdown it sending doesn't signal?

Second point, well I think 0-length is the correct opt-out mechanism, which is exactly to avoid ambiguity. IMO, a node signaling upfront_shutdown should opt-out or we may assume it's a buggy one ?

Reply:
Oops, you're right, sorry, still reading the wrong side of the spec.